### ISSUE TITLE
Code signing explanation

### ISSUE THREAD

**Issue by tomtaylor** [kgn/LaunchAtLoginHelper#1]
This looks great. 

Can you explain about how you're handling provisioning profiles and code signing across the main application and the helper? Every combination I've tried seems to have a flaw.

**Comment by kgn** [comment]
Hmm I am not sure, I haven't run into issues with provisioning profiles on the Mac. I'll describe what I have setup which might help:

In _Organizer_ for _My Mac_ I only have one provisioning profile, which is _Mac Team Provisioning Profile_:
![](http://f.cl.ly/items/2k1s410c2t2y3W3G3V2I/Screen%20Shot%202012-05-20%20at%2012.55.38%20PM.png)

Then the code signing for my main project is:
![](http://f.cl.ly/items/0H3A1B3L111r1l1n4610/codesign.png)

And the code signing for _LaunchAtLoginHelper_ app is unchanged and looks the same as in the screenshot above.

Hope this helps.

**Comment by tomtaylor** [comment]
Thanks - are the helper and the main application signed with the same key? Does your application pass Mac App Store validation?

**Comment by kgn** [comment]
Yes, I've submitted two apps that have been approved with this helper app:
- [Play by Play](http://itunes.apple.com/us/app/play-by-play/id474543724?mt=12)
- [ShellTo](http://itunes.apple.com/us/app/shellto/id418075847?mt=12)

**Comment by tomtaylor** [comment]
Thanks, that's good to know. I'm still a bit unsure about the code signing: did you use a separate provisioning profile on a separate bundle ID for the main app and the helper?

**Comment by kgn** [comment]
I used the same profile I think, I didn't have to do anything special with the provisioning profiles so I assume they both use the same one. Have you been able to submit the app yet?

**Comment by codler** [comment]
+1
Do you know if this work in mountain lion?

**Comment by holgersindbaek** [comment]
I'm getting this error when I submit my app to the app store, using the this otherwise awesome library:

```
Invalid Provisioning Profile Location - The provisioning profile for your Mac OS X app must be located in the
```

Contents directory of the main app bundle. A provisioning profile is optional, but you cannot submit more than one.

Anyone having any experience with that?

**Comment by kgn** [comment]
Hmm, I'm sorry to say that I have not submitted a new build to the app store in a while so I am not sure if anything has changed. Maybe make sure the same provisioning profiles are being used by your app and LaunchAtLoginHelper.

**Comment by holgersindbaek** [comment]
The same provisioning profile are being used.

I ended up solving it by removing the embedded profile in the helper app:

```
http://blog.mcohen.me/2012/05/20/sandboxing-revisited/
```

Doesn't seem like an optimal solution though.

**Comment by kgn** [comment]
Glad you found a solution! I'll have to investigate this more the next time I submit an update.

**Comment by holgersindbaek** [comment]
Would be awesome. It's the best launch helper I've found. Really nicely done!
